hass_nabucasa/remote.py

```diff
@@ -524,3 +535,90 @@ async def _certificate_handler(self) -> None:
 
         _LOGGER.debug("Stopping Remote UI loop")
         await self.close_backend()
+
+    async def _check_cname(self, hostname: str) -> list[str]:
+        """Get CNAME records for hostname."""
+        try:
+            return await cloud_api.async_resolve_cname(self.cloud, hostname)
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.error("Can't resolve CNAME for %s", hostname)
+        return []
+
+    async def _custom_domain_dns_configuration_is_valid(
+        self, instance_domain: str, custom_domain: str
+    ) -> bool:
+        """Validate custom domain."""
+        # Check primary entry
+        if instance_domain not in await self._check_cname(custom_domain):
+            return False
+
+        # Check LE entry
+        if f"_acme-challenge.{instance_domain}" not in await self._check_cname(
+            f"_acme-challenge.{custom_domain}"
+        ):
+            return False
+
+        return True
+
+    async def _should_renew_certificates(self) -> bool:
+        """Check if certificates should be renewed."""
+        bad_alias = []
+
+        if TYPE_CHECKING:
+            assert self._acme is not None
+            assert self.instance_domain is not None
+
+        if not self._acme.certificate_available:
+            return True
+
+        if self._acme.expire_date is None:
+            return True
+
+        if self._acme.expire_date > (
+            utils.utcnow() + timedelta(days=RENEW_IF_EXPIRES_DAYS)
+        ):
+            return False
+
+        check_alias = [
+            domain for domain in self._acme.domains if domain != self.instance_domain
+        ]
+
+        if len(check_alias) == 0:
+            return True
+
+        # Check if defined alias is still valid:
+        for alias in check_alias:
+            # Check primary entry
+            if not await self._custom_domain_dns_configuration_is_valid(
+                self.instance_domain, alias
+            ):
+                bad_alias.append(alias)
+
+        if len(bad_alias) == 0:
+            # No bad configuration detected
+            return True
+
+        if self._acme.expire_date > (
+            utils.utcnow() + timedelta(days=WARN_RENEW_FAILED_DAYS)
+        ):
+            await self.cloud.client.async_create_repair_issue(
+                identifier=f"warn_bad_custom_domain_configuration_{self._acme.expire_date.timestamp()}",
+                translation_key="warn_bad_custom_domain_configuration",
+                placeholders={"custom_domains": ",".join(bad_alias)},
+                severity="warning",
+            )
+            return False
+
+        # Recreate the acme client with working domains
+        await self.cloud.client.async_create_repair_issue(
```

Comment:
What are we asking users to do here?

Reply:
Currently nothing, this is a no-op untill its implemented in core.

But the idea here is to notify the user that we had to create a new certificate without their defined custom domain, and they would have to remove it from the account page, or fix the DNS config of that domain